script.embuary.info/resources/lib/tmdb_video.py

```diff
@@ -0,0 +1,206 @@
+#!/usr/bin/python
+# coding: utf-8
+
+########################
+
+import sys
+import xbmc
+import xbmcgui
+import requests
+
+from resources.lib.helper import *
+from resources.lib.tmdb_utils import *
+
+########################
+
+class TMDBVideos(object):
+    def __init__(self,call_request):
+        self.result = {}
+        self.call = call_request['call']
+        self.tmdb_id = call_request['tmdb_id']
+        self.local_movies = call_request['local_movies']
+        self.local_shows = call_request['local_shows']
+        self.movie = get_bool(self.call,'movie')
+        self.tvshow = get_bool(self.call,'tv')
+
+        if self.tmdb_id:
+            cache_key = str(call_request) + DEFAULT_LANGUAGE + COUNTRY_CODE
+            self.details = get_cache(cache_key)
+
+            if not self.details:
+                self.details = tmdb_item_details(self.call,self.tmdb_id,append_to_response='release_dates,content_ratings,external_ids,credits,videos,translations')
+                if self.details:
+                    write_cache(cache_key,self.details)
+
+            if not self.details:
+                return
+
+            self.created_by = self.details['created_by'] if self.details.get('created_by') else ''
+            self.cast = self.details['credits']['cast']
+            self.crew = self.details['credits']['crew']
+            self.videos = self.details['videos']['results']
+            self.details['crew'] = self.crew
+
+            self.result['details'] = self.get_details()
+            self.result['cast'] = self.get_cast()
+            self.result['crew'] = self.get_crew()
+            self.result['similar'] = self.get_similar()
+            self.result['youtube'] = self.get_yt_videos()
+            self.result['images'] = self.get_images()
+
+    def __getitem__(self, key):
+        try:
+            value = self.result[key]
+            return value
+        except KeyError:
```

Comment:
This could be better. self.result.get(key, "")

Reply:
changed